src/utils.rs

```diff
@@ -93,55 +86,86 @@ mod unsafe_code {
-    /// Create a new `RawFdContainer` for the given `RawFd`.
-    ///
-    /// The `RawFdContainer` takes ownership of the `RawFd` and closes it on drop.
-    ///
-    /// This function panics on non-unix systems.
-    pub fn new(fd: RawFd) -> RawFdContainer {
-        if cfg!(unix) {
+    impl RawFdContainer {
+        /// Create a new `RawFdContainer` for the given `RawFd`.
+        ///
+        /// The `RawFdContainer` takes ownership of the `RawFd` and closes it on drop.
+        ///
+        /// This function panics on non-unix systems.
+        pub fn new(fd: RawFd) -> Self {
             RawFdContainer(fd)
-        } else {
-            unimplemented!("RawFdContainer is only implemented on Unix-like systems");
+        }
+
+        /// Tries to clone the `RawFdContainer` creating a new FD
+        /// with `dup`. The new `RawFdContainer` will take ownership
+        /// of the `dup`ed version, whereas the original `RawFdContainer`
+        /// will keep the ownership of is FD.
+        pub fn try_clone(&self) -> nix::Result<Self> {
+            Ok(Self::new(nix::unistd::dup(self.0)?))
+        }
+
+        /// Get the `RawFd` out of this `RawFdContainer`.
+        ///
+        /// This function would be an implementation of `IntoRawFd` if that were possible. However, it
+        /// causes a conflict with an `impl` from libcore...
+        pub fn into_raw_fd(self) -> RawFd {
+            let fd = self.0;
+            forget(self);
+            fd
+        }
+
+        /// Consumes the `RawFdContainer` and closes the wrapped FD with
+        /// the `close` system call.
+        ///
+        /// This is similar to dropping the `RawFdContainer`, but it allows
+        /// the caller to handle errors.
+        pub fn close(self) -> nix::Result<()> {
+            let fd = self.into_raw_fd();
+            nix::unistd::close(fd)
         }
     }
 
-    /// Get the `RawFd` out of this `RawFdContainer`.
-    ///
-    /// This function would be an implementation of `IntoRawFd` if that were possible. However, it
-    /// causes a conflict with an `impl` from libcore...
-    pub fn into_raw_fd(self) -> RawFd {
-        let fd = self.0;
-        forget(self);
-        fd
+    impl<T: IntoRawFd> From<T> for RawFdContainer {
+        fn from(fd: T) -> Self {
+            Self::new(fd.into_raw_fd())
+        }
     }
-}
 
-#[cfg(unix)]
-impl<T: IntoRawFd> From<T> for RawFdContainer {
-    fn from(fd: T) -> RawFdContainer {
-        RawFdContainer::new(fd.into_raw_fd())
+    impl AsRawFd for RawFdContainer {
+        fn as_raw_fd(&self) -> RawFd {
+            self.0
+        }
     }
 }
 
-#[cfg(unix)]
-impl AsRawFd for RawFdContainer {
-    fn as_raw_fd(&self) -> RawFd {
-        self.0
-    }
+#[cfg(not(unix))]
+mod raw_fd_container {
+    /// A simple wrapper around RawFd that closes the fd on drop.
+    ///
+    /// On non-unix systems, or when the `allow-unsafe-code` is disabled,
+    /// this type is empty and does not provide any method.
```

Comment:
Please copy the doc from the `unix` version of `RawFdContainer` (this still talks about `allow-unsafe-code`)

Reply:
Fixed